server/src/lib/storage/filesystem.js

```diff
@@ -0,0 +1,51 @@
+'use strict';
+
+const storage = module.exports = {};
+const databaseDir = `${__dirname}/data`;
+const fs = require('fs');
+
+
+
+storage.save = data => {
+  return new Promise((resolve, reject) => {
+    if (!data.id) {
+      reject('Error: no id provided on data');
+    }
+    
+    let file = `${databaseDir}/${data.id}.json`;
+    let text = JSON.stringify(data);
+    fs.writeFile(file, text, (err => {
+      if (err) { reject(err); }
+      resolve(data);
+    }));
+  });
+};
+
+storage.get = id => {
+  return new Promise((resolve, reject) => {
+    if (!id) { reject('ERROR: no id provided in data'); }
+
+    let file = `${databaseDir}/${id}.json`;
+
+    fs.readFile(file, (err, data) => {
+      if (err) reject(err);
+      let obj = JSON.parse(data.toString());
+      resolve(obj);
+    });
+  });
+};
+
+storage.delete = (res, id) => {

```

Comment:
Passing the response object all the way down to our persistence layer is not necessary. Instead, we can just resolve with nothing passed, or resolve with a success message (as string or object or whatever).